scanner/src/main/kotlin/experimental/NestedProvenanceScanResult.kt

```diff
@@ -48,4 +55,77 @@ data class NestedProvenanceScanResult(
      * [nestedProvenance].
      */
     fun isComplete(): Boolean = getProvenances().all { scanResults[it]?.isNotEmpty() == true }
+
+    /**
+     * Merge the nested [ScanResult]s into one [ScanResult] per used scanner, using the root of the [nestedProvenance]
+     * as provenance. This is used to transform this class into the format currently used by [OrtResult].
+     * When merging multiple [ScanSummary]s the earliest start time will be used as the new start time, and the latest
+     * end time will be used as the end time. Because the [ScanSummary] does not contain the checksums of the individual
+     * files, no package verification code can be calculated.
+     */
+    fun merge(): List<ScanResult> {
+        val allScanners = scanResults.values.flatMapTo(mutableSetOf()) { results -> results.map { it.scanner } }
+
+        return allScanners.map { scanner ->
+            val scanResultsForScanner = scanResults.mapValues { (_, results) ->
+                results.filter { it.scanner == scanner }
+            }
+
+            val allScanResults = scanResults.values.flatten()
+
+            val startTime = allScanResults.minByOrNull { it.summary.startTime }?.summary?.startTime ?: Instant.now()
+            val endTime = allScanResults.maxByOrNull { it.summary.endTime }?.summary?.endTime ?: Instant.now()
```

Comment:
Nit: Shouldn't these use exactly the same Instant instance as a fallback?

Reply:
I am using `startTime` as a fallback now, that should achieve this.